### ISSUE TITLE
Flake8 rule 'E999' reported an invalid point

### ISSUE THREAD

**Issue by twof** [AtomLinter/linter-flake8#297]
Flake8 reported an invalid point for the rule `E999`, with the messge `SyntaxError: unexpected EOF while parsing`.

```
from linked_list import Linked_List


class HashTable(object):
    # O(n)
    def __init__(self, init_size=8):
        """Initialize this hash table with the given initial size"""
        self.buckets = [Linked_List() for i in range(init_size)]
        self.current_index = 0
        self.current_node = self.buckets[0].head

    # O(1)
    def __repr__(self):
        """Return a string representation of this hash table"""
        return 'HashTable({})'.format(self.length())

    # O(1)
    def __iter__(self):
        return self

    # O()
    def __next__(self):
        node_to_return = self.current_node
        if self.current_index == len(self.buckets)-1:
            raise StopIteration
        else:
            try:
                self.current_node = self.buckets[self.current_index].__next__()
            except StopIteration:
                self.current_index += 1
                return node_to_return

    # O(1)
    def _bucket_index(self, key):
        """Return the bucket index where the given key would be stored"""
        return hash(key) % len(self.buckets)

    # O(n)
    def length(self):
        """Return the length of this hash table by traversing its buckets"""
        total = 0

        for list in self.buckets:
            total += list.length()

        return total

    # O(n)
    def contains(self, key):
        """Return True if this hash table contains the given key, or False"""
        bucket = self.buckets[self._bucket_index(key)]

        if bucket.find(lambda item: item[0] == key) is not None:
            return True
        else:
            return False

    # O(n)
    def get(self, key):
        """Return the value associated with the given key, or raise KeyError"""
        bucket = self.buckets[self._bucket_index(key)]
        found_item = bucket.find(lambda item: item[0] == key)

        if found_item is not None:
            return found_item[1]
        else:
            raise KeyError

    # O(n)
    def set(self, key, value):
        """Insert or update the given key with its associated value"""
        bucket = self.buckets[self._bucket_index(key)]
        bucket.upsert_first([key, value], lambda item: item[0] == key)

    # O(n)
    def delete(self, key):
        """Delete the given key from this hash table, or raise KeyError"""
        bucket = self.buckets[self._bucket_index(key)]

        try:
            bucket.delete(key, lambda item: item[0] == key)
        except ValueError as e:
            raise KeyError

    # O(n + b)
    def keys(self):
        """Return a list of all keys in this hash table"""
        key_list = []

        for bucket in self.buckets:
            bucket_list = bucket.as_list(lambda data: data[0])
            key_list.extend(bucket_list)

        return key_list

    # O(n + b)
    def values(self):
        """Return a list of all values in this hash table"""
        val_list = []

        for bucket in self.buckets:
            bucket_list = bucket.as_list(lambda data: data[1])
            val_list.extend(bucket_list)

        return val_list

    # defaults to buckets = number of items in the table
    def expand_table(self, size=None):
        if size is None:
            

```

Debug information:
Atom version: 1.12.4
Flake8 version: `3.0.4 (pycodestyle: 2.0.0, pyflakes: 1.2.3, mccabe: 0.5.2, flake8-docstrings: 1.0.2, pydocstyle: 1.0.0) CPython 3.4.1 on Darwin`

**Comment by Arcanemagus** [comment]
Please update your `flake8` version (`pip install -U flake8`). This particular error should be fixed in `flake8>=3.2.1`.

**Comment by twof** [comment]
Thanks!
